### PR TITLE
Move mockredis to test requirements

### PR DESCRIPTION
**Pull request by blueyed**
It is not necessary to be installed typically.

### CODE REVIEW

**Comment by blueyed** [comment]
It might make sense to remove the `DEFENDER_MOCK_REDIS` setting altogether - which makes this kind of supported.
Or, when leaving it, it should be documented that mockredis needs to get installed, too.

File: defender/connection.py [hunk 1]
```diff
@@ -13,14 +12,18 @@
...
                           ' settings.'
 
 
 def get_redis_connection():
     """ Get the redis connection if not using mock """
     if config.MOCK_REDIS:  # pragma: no cover
+        global MOCKED_REDIS
+        if MOCKED_REDIS is None:
+            import mockredis
+            MOCKED_REDIS = mockredis.mock_strict_redis_client()
```
Comment (kencochrane) [inline defender/connection.py]: We can probably make this even more simple.

```
if config.MOCK_REDIS:
    import mockredis
    return mockredis.mock_strict_redis_client()
```

Then we can remove the `MOCKED_REDIS` variable all together what do you think?

Comment (blueyed) [reply defender/connection.py]: Makes sense.

Please feel free to pick up this PR and/or do it from scratch - I've decided to not use django-defender, so will not update it myself.

**Comment by kencochrane** [comment]
closing, and replacing with #119

**Pull request closed without merging**